### PR TITLE
Templatize ClosedInterval

### PR DESCRIPTION
**Pull request by karupayun**
I need to use with an integer type so I'm proposing to templatize it.

### CODE REVIEW

File: src/OrbitGl/CoreMath.h [hunk 1]
```diff
@@ -29,14 +29,13 @@ using Color = gte::Vector4<unsigned char>;
 
 namespace orbit_gl {
 
+template <typename T>
 struct ClosedInterval {
   [[nodiscard]] bool Intersects(const ClosedInterval& closed_interval) const {
     return this->min <= closed_interval.max && this->max >= closed_interval.min;
   }
 
-  [[nodiscard]] bool Contains(float value) const {
```
Comment (beckerhe) [inline src/OrbitGl/CoreMath.h:37]: This struct would benefit from a static_assert that limits it to arithmetic types:

```cpp
static_assert(std::is_arithmetic_v<T>, "T needs to be an arithmetic type (integral or floating point type).");
```

File: src/OrbitGl/CoreMathTest.cpp [hunk 2]
```diff
@@ -8,19 +8,19 @@
 
 namespace orbit_gl {
 
-const ClosedInterval kSmallClosedInterval{0, 3};
+const ClosedInterval<int> kSmallClosedInterval{0, 3};
```
Comment (beckerhe) [inline src/OrbitGl/CoreMathTest.cpp:11]: Your usage of ClosedInterval is float-only (so far) but the tests are all with `int`. What about adding a `TYPED_TEST_SUITE`/`TYPED_TEST`?

**Review by beckerhe — APPROVED** [review]
Otherwise LGTM

File: src/OrbitGl/CoreMath.h (hunk 1 above)
Comment (karupayun) [reply src/OrbitGl/CoreMath.h:37]: In the constructor?

Comment (beckerhe) [reply src/OrbitGl/CoreMath.h:37]: Just in the struct declaration - as if it was declaring a member variable.

**Review by beckerhe — APPROVED** [review]
LGTM. Thx. Cool!

**Pull request merged**